backend/src/main/java/ca/gov/dtsstn/cdcp/api/service/UserService.java

```diff
@@ -134,4 +134,25 @@ private Predicate<ConfirmationCodeEntity> byCode(String code) {
 		return confirmationCode -> confirmationCode.getCode().equals(code);
 	}
 
+	private Predicate<ConfirmationCodeEntity> notExpired() {
+		return confirmationCode -> confirmationCode.getExpiryDate().isAfter(Instant.now());
+	}
+
+    public Boolean verifyConfirmationCode(String code, User user){
+		Assert.hasText(code, "code is required; it must not be null or blank");
+		Assert.notNull(user, "user is required; it must not be null");
+		if (user.getConfirmationCodes().stream().map(confirmationCodeMapper::toEntity).filter(notExpired()).noneMatch(byCode(code))) {
+    		return false;
+    	}
+        return true;
+    }
+
+	public void setEmailValidated(String userId) {
```

Comment:
I would probably roll this method into the verify method above and name it `verifyEmail(..)` since searching for the code, setting verified=true, and deleting all other confirmation codes will always be an atomic operation.

Reply:
Sounds reasonable, implemented in commit 2cebd2d6f0685660fa512c66c4a5da16869070e1.